src/sdk/resources/resource.ts

```diff
@@ -41,6 +28,25 @@ export class Resource<TResource> {
 
 		return Object.assign({}, lhs, rhs);
 	}
+}
+
+export class Resource<TResource> extends BaseResource {
+	constructor(protected httpClient: IHttpClient, protected singularName: string, protected pluralName: string = undefined, 
```

Comment:
Does this work with `protected` qualifier on this field since we technically just want to use the base `httpClient`? Shouldn't this just be `httpClient: IHttpClient` so that it doesn't attempt to assign it to any local field since we already do that as a result of the call to `super(httpClient);`?

Reply:
It does work (I checked on a cmd that uses resource before I committed); however, I see why this makes sense since it inherits the BaseResource constructor. `protected` removed.